content/en/blog/extend-falco-inputs-with-a-plugin-distribute/index.md

```diff
@@ -0,0 +1,495 @@
+        run: make
+        working-directory: tools/falcoctl
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: plugin
+      - name: Build the plugin
+        run: make build
+        working-directory: plugin
+      - id: StringRepoName
+        uses: ASzc/change-string-case-action@v5
+        with:
+          string: ${{ github.repository }}
+      - name: Upload OCI artifacts to GitHub packages
+        run: |
+              MAJOR=$(echo ${{ github.ref_name }} | cut -f1 -d".")
+              MINOR=$(echo ${{ github.ref_name }} | cut -f1,2 -d".")
+              DIR=$(pwd)
+
+              cd plugin/
+              $DIR/tools/falcoctl/falcoctl registry push \
+              ${{ env.OCI_REGISTRY }}/${{ steps.StringRepoName.outputs.lowercase }}/plugin/${{ env.PLUGIN_NAME }}:${{ github.ref_name }} \
+              --config /dev/null \
+              --type plugin \
+              --version "${{ github.ref_name }}" \
+              --tag latest --tag $MAJOR --tag $MINOR \
+              --platform linux/amd64 \
+              --requires plugin_api_version:2.0.0 \
+              --depends-on ${{ env.PLUGIN_NAME }}-rules:${{ github.ref_name }} \
+              --name ${{ env.PLUGIN_NAME }} \
+              lib${{ env.PLUGIN_NAME }}.so
+    
+              cd rules/
+              $DIR/tools/falcoctl/falcoctl registry push \
+              ${{ env.OCI_REGISTRY }}/${{ steps.StringRepoName.outputs.lowercase }}/ruleset/${{ env.PLUGIN_NAME }}:${{ github.ref_name }} \
+              --config /dev/null \
+              --type rulesfile \
+              --version "${{ github.ref_name }}" \
+              --tag latest --tag $MAJOR --tag $MINOR \
+              --depends-on ${{ env.PLUGIN_NAME }}:${{ github.ref_name }} \
+              --name ${{ env.PLUGIN_NAME }}-rules \
+              ${{ env.PLUGIN_NAME }}_rules.yaml
+        env:
+          FALCOCTL_REGISTRY_AUTH_BASIC: ${{ env.OCI_REGISTRY }},${{ github.repository_owner }},${{ secrets.GITHUB_TOKEN }}
+
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Setup Golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          version: latest
+          args: release --clean --timeout 120m
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          LDFLAGS: "-buildmode=c-shared"
+          GOPATH: /home/runner/go
+```
+
+Replace `PLUGIN_NAME` with the name of your plugin.
+
+## index.yaml file
+
+This file is used by `falcoctl` to know where to download the plugins and rules. We'll create our own file to allow anyone to use it locally.
+
+```yaml
+- name: {PLUGIN_NAME}
+  type: plugin
+  registry: ghcr.io
+  repository: {OWNER_NAME}/{REPO_NAME}/plugin/{PLUGIN_NAME}
+  description: Reads events from Docker Engine
```

Comment:
Suddenly it is not a generic plugin anymore. Is this on purpose?

Reply:
Wrong copy/paste